### PR TITLE
Refactoring of Restart Cluster Server and Add the Certificate Rotation

### PR DESCRIPTION
**Pull request by est-suse**
#### Proposed Changes ####
Add rotation certificate check and refactor the restart check

Same as the PR https://github.com/k3s-io/k3s/pull/7097

#### Types of Changes ####
Test

#### Verification ####
Run the test

### CODE REVIEW

File: tests/e2e/validatecluster/validatecluster_test.go [hunk 1]
```diff
@@ -133,7 +134,7 @@ var _ = Describe("Verify Basic Cluster Creation", Ordered, func() {
 		}, "240s", "5s").Should(ContainSubstring("test-loadbalancer"), "failed cmd: "+cmd)
 	})
 
-	It("Verifies Restart", func() {
+	/*It("Verifies Restart", func() {
```
Comment (brandond) [inline tests/e2e/validatecluster/validatecluster_test.go:137]: delete it if you want it gone, don't just comment it out

Comment (est-suse) [reply tests/e2e/validatecluster/validatecluster_test.go]: Yes thanks

**Comment by est-suse** [comment]
> You will need to change the function for start /stop from k3s to rke2... also you should test this locally, Rke2 restarts much slower than k3s so the timeouts may need to be larger.

Done, Thanks, could you please take a look again?

File: tests/e2e/validatecluster/validatecluster_test.go [hunk 2]
```diff
@@ -247,8 +199,132 @@ var _ = Describe("Verify Basic Cluster Creation", Ordered, func() {
...
+				podsRunningAr := 0
+				for _, pod := range pods {
+					if strings.Contains(pod.Name, "test-daemonset") && pod.Status == "Running" && pod.Ready == "1/1" {
+						podsRunningAr++
+					}
+				}
+				g.Expect(len(nodes)).Should((Equal(podsRunningAr)), "Daemonset pods are not running after the restart")
+			}, "1120s", "5s").Should(Succeed())
+		})
+	})
```
Comment (rancher-max) [inline tests/e2e/validatecluster/validatecluster_test.go:245]: I don't know how this is working -- the first `It` function has the `defer` statement to remove the daemonset workload, but the next one then expects it to be present. I guess this is the same code in k3s and it is working there too, but how? @dereknola any ideas?

**Review by rancher-max — COMMENTED** [review]
This LGTM as long as it works with the already-defined timeouts, which it seems it does since CI is passing.

**Review by brandond — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Review by ShylajaDevadiga — APPROVED** [review]

**Pull request merged**